### PR TITLE
[FrameworkBundle][config cmd] initialize extension.

### PR DESCRIPTION
**Pull request by aitboudad**
| Q | A |
| --- | --- |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Fixed tickets | #13196, #11037 |
| Tests pass? | yes |
| License | MIT |

### CODE REVIEW

**Comment by fabpot** [comment]
Any way to test this?

**Comment by aitboudad** [comment]
why not, I'll try :)

**Comment by aitboudad** [comment]
Test added

File: src/Symfony/Bundle/FrameworkBundle/FrameworkBundle.php [hunk 1]
```diff
@@ -82,7 +82,7 @@ public function build(ContainerBuilder $container)
         $container->addCompilerPass(new FragmentRendererPass(), PassConfig::TYPE_AFTER_REMOVING);
         $container->addCompilerPass(new SerializerPass());
 
-        if ($container->getParameter('kernel.debug')) {
+        if ($container->hasParameter('kernel.debug') && $container->getParameter('kernel.debug')) {
```
Comment (stof) [inline src/Symfony/Bundle/FrameworkBundle/FrameworkBundle.php]: this looks wrong. The Kernel class ensures that the kernel parameters are always set in the container when calling the bundle classes. You should not remove this assumption IMO.

File: src/Symfony/Bundle/FrameworkBundle/Tests/Functional/Bundle/TestBundle/TestBundle.php [hunk 2]
```diff
@@ -12,7 +12,17 @@
 namespace Symfony\Bundle\FrameworkBundle\Tests\Functional\Bundle\TestBundle;
 
 use Symfony\Component\HttpKernel\Bundle\Bundle;
-
```
Comment (stof) [inline src/Symfony/Bundle/FrameworkBundle/Tests/Functional/Bundle/TestBundle/TestBundle.php]: please keep the empty line between use statements and the class

File: src/Symfony/Bundle/FrameworkBundle/Command/AbstractConfigCommand.php [hunk 3]
```diff
@@ -48,7 +50,8 @@ protected function findExtension($name)
             // input is bundle name
 
             if (isset($bundles[$name])) {
-                $extension = $bundles[$name]->getContainerExtension();
+                $bundle = $bundles[$name];
```
Comment (jakzal) [inline src/Symfony/Bundle/FrameworkBundle/Command/AbstractConfigCommand.php]: `$bundle` is not initialized before, but you rely on it in the outer scope. I'd suggest to initialize it to `null`, just like we do with the [`$extension`](https://github.com/aitboudad/symfony/blob/config_cmd/src/Symfony/Bundle/FrameworkBundle/Command/AbstractConfigCommand.php#L45).

Comment (aitboudad) [inline src/Symfony/Bundle/FrameworkBundle/Command/AbstractConfigCommand.php]: fixed

File: src/Symfony/Bundle/FrameworkBundle/Command/AbstractConfigCommand.php [hunk 4]
```diff
@@ -83,4 +89,14 @@ public function validateConfiguration(ExtensionInterface $extension, $configurat
...
     }
+
+    private function initializeExtension($bundle, $extension)
+    {
+        $parameters = $this->getContainer()->getParameterBag()->all();
+        $container = new ContainerBuilder(new ParameterBag($parameters));
+        $container->registerExtension($extension);
+
+        $bundle = clone $bundle;
+        $bundle->build($container);
```
Comment (fabpot) [inline src/Symfony/Bundle/FrameworkBundle/Command/AbstractConfigCommand.php]: The whole code looks like voodoo to me :) A comment that explains what it does would probably help.

Comment (aitboudad) [inline src/Symfony/Bundle/FrameworkBundle/Command/AbstractConfigCommand.php]: done

File: src/Symfony/Bundle/FrameworkBundle/Tests/Functional/ConfigDumpReferenceCommandTest.php [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+    }
+
+    public function testDumpBundleName()
+    {
+        $tester = $this->createCommandTester();
+        $ret = $tester->execute(
+            array(
+                'name'    => 'TestBundle',
+            )
+        );
```
Comment (fabpot) [inline src/Symfony/Bundle/FrameworkBundle/Tests/Functional/ConfigDumpReferenceCommandTest.php]: Can you put everything on one line?

**Comment by stof** [comment]
@aitboudad FYI, we are not notified when you push new commits, so it is good to add a comment saying you updated the PR. Otherwise, it is likely that we won't come back to review it (there has been a 16 days delay here)

File: src/Symfony/Bundle/FrameworkBundle/Command/AbstractConfigCommand.php [hunk 6]
```diff
@@ -83,4 +89,16 @@ public function validateConfiguration(ExtensionInterface $extension, $configurat
...
+
+    private function initializeExtension($bundle, $extension)
+    {
+        // Re-build bundle manually to initialize DI extensions that can be extended by other bundles in their build() method
+        // as this method is not called when the container is loaded from the cache.
+        $parameters = $this->getContainer()->getParameterBag()->all();
+        $container = new ContainerBuilder(new ParameterBag($parameters));
+        $container->registerExtension($extension);
+
+        $bundle = clone $bundle;
```
Comment (stof) [inline src/Symfony/Bundle/FrameworkBundle/Command/AbstractConfigCommand.php]: what if ``$bundle`` is null ?

File: src/Symfony/Bundle/FrameworkBundle/Command/AbstractConfigCommand.php [hunk 7]
```diff
@@ -83,4 +89,16 @@ public function validateConfiguration(ExtensionInterface $extension, $configurat
             throw new \LogicException(sprintf('Configuration class "%s" should implement ConfigurationInterface in order to be dumpable', get_class($configuration)));
         }
     }
+
+    private function initializeExtension($bundle, $extension)
```
Comment (stof) [inline src/Symfony/Bundle/FrameworkBundle/Command/AbstractConfigCommand.php]: arguments should be typehinted

File: src/Symfony/Bundle/FrameworkBundle/Command/AbstractConfigCommand.php (hunk 4 above)
Comment (stof) [inline src/Symfony/Bundle/FrameworkBundle/Command/AbstractConfigCommand.php]: This won't solve the issue entirely actually. The whole point of extensions having a config which depends on the bundle build method is to allow *other* bundles than themselves to apply changes on it (for instance, DoctrineBundle registers the support of the entity type)

**Comment by aitboudad** [comment]
@stof didn't know about that so thank you for the clarification :) , 
I applied the comments, thanks!

**Comment by fabpot** [comment]
Thank you @aitboudad.

**Pull request closed without merging**